src/main/kotlin/org/rust/ide/highlight/RsRainbowVisitor.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.highlight
+
+import com.intellij.codeInsight.daemon.RainbowVisitor
+import com.intellij.codeInsight.daemon.impl.HighlightVisitor
+import com.intellij.openapi.editor.colors.TextAttributesKey
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsPatBinding
+import org.rust.lang.core.psi.RsPath
+
+class RsRainbowVisitor : RainbowVisitor() {
+    override fun suitableForFile(file: PsiFile): Boolean = file is RsFile
+
+    override fun clone(): HighlightVisitor = RsRainbowVisitor()
+
+    override fun visit(element: PsiElement) {
+        when (element) {
+            is RsPatBinding -> processPatBindingVisitor(element)
+            is RsPath -> processPathVisitor(element)
+        }
+    }
+
+    private fun processPatBindingVisitor(ref: RsPatBinding) {
+        addInfo(ref, ref.referenceNameElement, "${ref.referenceName}_${ref.textOffset}")
+    }
+
+    private fun processPathVisitor(element: RsPath) {
+        val ref = element.reference.resolve() ?: return
+        when (ref) {
+            is RsPatBinding ->
+                addInfo(ref, element, "${ref.referenceName}_${ref.textOffset}", RsColor.IDENTIFIER.textAttributesKey)
```

Comment:
Why do we need to include a `textOffest` here? It causes all of the colors change after the char is typed, because `textOffset`s of all the following identifiers are shifted.

Local typing should not change colors of other variables.